modules/dhcp/dhcp_api.rb

```diff
@@ -31,7 +31,7 @@ def load_subnet_data
   get "/?" do
     begin
       content_type :json
-      server.subnets.map{|s| {:network => s.network, :netmask => s.netmask, :options => s.options}}.to_json
+      server.subnets.map{|s| {:network => s.network, :netmask => s.netmask, :options => s.options, :prefix => s.prefix}}.to_json
```

Comment:
I think api controller needs:
 * change the interface to providers -- drop ```load_subnet``` and ```load_subnet_data``` calls. If you look at the recent and pending changes you'll notice that those aren't being used. In other words, one call to provider per operation.
 * keep ipv4 and ipv6 calls separate (by possibly prefixing the latter with 'ipv6'?)
  * You may have broken current (and older) versions of foreman with this implementation of ```get /```
  * create/delete/free ip calls are going to be different for ipv4 and ipv6, better to keep them separate to avoid guessing which needs to be called
 * Similarly for record retrieval: At least dhcp option calls are going to be ipv4/ipv6-specific, probably easier to keep those separate